[critical] Java SDK: ApiClient NPE crashes all examples for OSS users — no auth keys needed for OSS

## Problem

Any first-time Java SDK user connecting to a plain OSS Conductor server hits an immediate `NullPointerException` before executing a single line of example code.

This was discovered during a live zero-to-one run of the Java SDK examples against a local OSS Conductor server (started with `conductor server start`).

## Reproduction

```bash
export CONDUCTOR_SERVER_URL=http://localhost:8080/api
# no auth keys — correct for OSS
./gradlew :examples:compileJava
java -cp <examples-classpath> com.netflix.conductor.sdk.examples.helloworld.Main
```

**Error:**
```
Caused by: java.lang.NullPointerException: Cannot invoke "String.trim()"
because the return value of "java.lang.System.getenv(String)" is null
    at io.orkes.conductor.client.ApiClient$ApiClientBuilder.applyEnvVariables(ApiClient.java:186)
```

Root cause: `ApiClient.applyEnvVariables()` calls `.trim()` on the result of `System.getenv("CONDUCTOR_SERVER_AUTH_KEY")` without null-checking it first. Since OSS users don't set auth keys, both the primary and legacy env vars are null — crash.

Technical issue filed: **conductor-oss/java-sdk#94**

## Additional Zero-to-One Issues Found (same session)

Two more high-priority issues found during the same live run:

- **conductor-oss/java-sdk#95** — `examples` module has no `run` task; `./gradlew :examples:run` fails immediately with "task not found". No README shows how to execute any example.
- **conductor-oss/java-sdk#96** — Most examples (including `helloworld/Main.java`) use the Orkes-specific `ApiClient` that requires auth credentials. Only the `gettingstarted` package uses the base `ConductorClient` that works with OSS. This is undocumented.

## What Works

The `com.netflix.conductor.gettingstarted` examples (`CreateWorkflow`, `StartWorkflow`, `HelloWorker`) do work correctly against an OSS server when invoked manually with a classpath. Workflows were confirmed to complete successfully.

## User Impact

A developer following any Java SDK documentation hits a crash before getting any output. This blocks zero-to-one onboarding entirely.

--- a/examples/old/src/main/java/com/netflix/conductor/sdk/examples/helloworld/Main.java
+++ b/examples/old/src/main/java/com/netflix/conductor/sdk/examples/helloworld/Main.java
@@ -16,15 +16,15 @@ import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
+import com.netflix.conductor.client.http.ConductorClient;
 import com.netflix.conductor.sdk.examples.helloworld.workflowdef.GreetingsWorkflow;
 import com.netflix.conductor.sdk.workflow.executor.WorkflowExecutor;
-
-import io.orkes.conductor.sdk.examples.util.ClientUtil;
 
 public class Main {
 
     public static void main(String[] args) throws ExecutionException, InterruptedException, TimeoutException {
-        var workflowExecutor = new WorkflowExecutor(ClientUtil.getClient(), 10);
+        String serverUrl = System.getenv().getOrDefault("CONDUCTOR_SERVER_URL", "http://localhost:8080/api");
+        var workflowExecutor = new WorkflowExecutor(new ConductorClient(serverUrl), 10);
         workflowExecutor.initWorkers("com.netflix.conductor.sdk.examples.helloworld.workers");
         var workflowCreator = new GreetingsWorkflow(workflowExecutor);
         var simpleWorkflow = workflowCreator.create();

